include/fmt/format-inl.h

```diff
@@ -423,6 +423,18 @@ class fp {
     lower.f <<= lower.e - upper.e;
     lower.e = upper.e;
   }
+
+  void compute_float_boundaries(fp& lower, fp& upper) const {
+    constexpr int min_normal_e = std::numeric_limits<float>::min_exponent -
+                                 std::numeric_limits<double>::digits;
```

Comment:
The smallest positive normal float is `2^-126`. `fp(2^-126)` is `fp(2^52, -178)`. float min_exponent is -125, double digits is 53: both of these numbers are off by one from what we need, but these 1s cancel each other out.